code/game/gamemodes/objectives/objective_heist.dm

```diff
@@ -11,10 +11,11 @@
 	for(var/datum/mind/possible_target in SSticker.minds)
 		if(possible_target != owner && ishuman(possible_target.current) && (possible_target.current.stat != DEAD) && (!possible_target.special_role))
 			possible_targets += possible_target
-			for(var/datum/job/role in SSjobs.get_by_path(roles))
-				if(possible_target.assigned_role == role.title)
-					priority_targets += possible_target
-					continue
+			if(length(roles))
+				for(var/datum/job/role in SSjobs.get_by_path(roles))
+					if(possible_target.assigned_role == role.title)
```

Comment:
`possible_target.assigned_role == /datum/job/chief_engineer.title`
needs initial maybe?

Reply:
`get_by_path` returns a datum.

Reply:
derp didn't see that